modules/mono/editor/bindings_generator.cpp

```diff
@@ -1504,37 +1515,48 @@ Error BindingsGenerator::_generate_cs_type(const TypeInterface &itype, const Str
 						"' for class '" + itype.name + "'.");
 	}
 
-	if (itype.is_singleton) {
-		// Add the type name and the singleton pointer as static fields
+	// Add native name static field and cached type.
 
-		output.append(MEMBER_BEGIN "private static GodotObject singleton;\n");
+	if (is_derived_type && !itype.is_singleton) {
+		output << MEMBER_BEGIN "private static readonly System.Type CachedType = typeof(" << itype.proxy_name << ");\n";
```

Comment:
(probably preexisting) but why use this static field instead of just inlining the `typeof(x)` at the use sites?

Reply:
I don't know, it seems it's only used with `_ConstructAndInitialize` in the constructor, so it could be replaced with `typeof(x)`. I left it since it's unrelated to this PR, maybe it could be removed in a separate PR.